pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -425,6 +426,20 @@ func (t *thirdPartyResourceDataEncoder) EncodeToStream(obj runtime.Object, strea
 		}
 		fmt.Fprintf(stream, template, t.kind+"List", strings.Join(dataStrings, ","))
 		return nil
+	case *versioned.InternalEvent:
+		event := &versioned.Event{}
+		err := versioned.Convert_versioned_InternalEvent_to_versioned_Event(obj, event, nil)
+		if err != nil {
+			return err
+		}
+
+		enc := json.NewEncoder(stream)
```

Comment:
@brendandburns I dont think this is possible as it currently stands e.g. no kind "WatchEvent" is registered for version "stable.sitepod.io/v1"